openapi.yaml

```diff
@@ -0,0 +1,712 @@
+          description: Too many requests
+        '500':
+          description: Internal server error _(service logo cannot be updated)_
+      security:
+        - subscription_key: []
+  /services/{serviceId}/key:
+    put:
+      tags:
+        - services-authorization
+      summary: Regenerate service key
+      description: |
+        Regenerate service key by `serviceId`. <br><br>_Note: This API operation needs the __Manage API key__ owned by the service user, otherwise `403 forbidden` will be returned to the caller._
+      operationId: regenerateServiceKey
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Service key type
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SubscriptionKeyTypePayload'
+        required: true
+      responses:
+        '200':
+          description: Successful operation _(service key regenerated)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SubscriptionKeys'
+        '400':
+          description: Invalid payload
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service key cannot be regenerated)_
+      security:
+        - subscription_key: []
+  /services/{serviceId}/actions/review:
+    patch: 
+      tags:
+        - service-actions
+      summary: Send service to review
+      description: |
+        Send service to review by `serviceId`. <br><br>_Note: This API operation needs the __Manage API key__ owned by the service user,
+        otherwise `403 forbidden` will be returned to the caller._
+      operationId: reviewService
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation _(service revirew taken in charge)_
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Todo'
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '404':
+          description: Not found
+        '412':
+          description: Precondition failed _(service status is incompatible with review action request)_
+        '429':
+          description: Too many requests
+        '500':
+          description: Internal server error _(service review cannot be taken in charge)_
+      security:
+        - subscription_key: []
+  /services/{serviceId}/actions/explain:
+    patch: 
```

Comment:
I am not very "happy" with the design of "actions" because they are not RESTful API.

Reply:
see comment above